src/main/java/com/eternalcode/parcellockers/gui/ParcelListGUI.java

```diff
@@ -53,30 +61,49 @@ public void showParcelListGUI(Player player) {
-                    .name(this.miniMessage.deserialize(meta.getName()))
-                    .lore(this.miniMessage.deserialize("&3UUID: ").append(Component.text(parcel.uuid().toString())),
-                            this.miniMessage.deserialize("&3Sender: ").append(Component.text(sender)),
-                            this.miniMessage.deserialize("&3Receiver: ").append(Component.text(receiver)),
-                            this.miniMessage.deserialize("&3Size: ").append(Component.text(meta.getSize().toString())),
-                            this.miniMessage.deserialize("&3Priority: ").append(Component.text(meta.isPriority() ? "&aYes" : "&cNo")),
-                            this.miniMessage.deserialize("&3Description: ").append(Component.text(meta.getDescription())),
-                            this.miniMessage.deserialize("&3Position: ").append(Component.text(meta.getDestinationLocker().getPosition().toString())),
-                            this.miniMessage.deserialize("&3Recipients: ").append(Component.text(meta.getRecipients().stream().map(Bukkit::getPlayer).map(HumanEntity::getName).toList().toString())))
-                    .flags(ItemFlag.HIDE_ATTRIBUTES)
-                    .asGuiItem());
+        this.parcelRepository.findAll().whenComplete((parcels, throwable) -> {
+            for (Parcel parcel : parcels) {
+                List<String> newLore = this.replaceParcelPlaceholders(parcel, parcelItem.getItemStack().getItemMeta().getLore());
+                parcelItem.getItemStack().getItemMeta().setLore(newLore);
+
+                gui.addItem(parcelItem);
+            }
+
+            gui.setItem(49, closeItem);
+            this.server.getScheduler().runTask(this.plugin, () -> gui.open(player));
+        });
+
+    }
+
+    public List<String> replaceParcelPlaceholders(Parcel parcel, List<String> lore) {
+        if (lore == null || lore.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ParcelLocker destination = this.parcelLockerRepository.findByUUID(parcel.destinationLocker()).join().get();
+        Formatter formatter = new Formatter();
+        formatter.register("{UUID}", parcel.uuid().toString());
+        formatter.register("{NAME}", parcel.name());
+        formatter.register("{SENDER}", this.server.getPlayer(parcel.sender()).getName());
+        formatter.register("{RECEIVER}", this.server.getPlayer(parcel.receiver()).getName());
+        formatter.register("{SIZE}", parcel.size().toString());
+        formatter.register("{PRIORITY}", parcel.priority() ? this.miniMessage.deserialize("&aYes") : this.miniMessage.deserialize("&cNo"));
+        formatter.register("{DESCRIPTION}", parcel.description());
+        formatter.register("{POSITION_X}", destination.position().x());
+        formatter.register("{POSITION_Y}", destination.position().y());
+        formatter.register("{POSITION_Z}", destination.position().z());
+        formatter.register("{RECIPIENTS}", parcel.recipients().stream()
```

Comment:
Możesz zapisać to w chain.
np.
new Formatter()
    .register("{UUID}", parcel.uuid().toString())
    .register("{NAME}", parcel.name());
    .register("{SENDER}", this.server.getPlayer(parcel.sender()).getName());
    .register("{RECEIVER}", this.server.getPlayer(parcel.receiver()).getName());